app/controllers/okta_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+require 'openid_auth'
+
+class OktaController < ApplicationController
+  include OpenidAuth::V0
+  skip_before_action :authenticate
+
+  def metadata
+    meta = OneLogin::RubySaml::Metadata.new
+    render xml: meta.generate(saml_settings), content_type: 'application/xml'
+  end
+
+  def fetch_mvi_profile(json_request)
+    # TODO: pull out info from Okta request for querying MVI. Put into helper func
+    okta_attributes = json_request.dig(:data, :assertion, :claims)
+    user_attributes = Hash.new
+    okta_attributes.each do |k, v|
+      user_attributes[k] = v[:attributeValues][0][:value]
+    end
+    user_attributes["authn_context"] = json_request[:data][:assertion][:authentication][:authnContext][:authnContextClassRef]
+    
+    binding.pry
+    user_identity = OpenidAuth::V0::MviUsersController.new.build_identity_from_attributes(user_attributes)
```

Comment:
this needs to get moved, but I was still having trouble getting this method to run after moving this controller inside the `OpenIdAuth` module

Reply:
I would  just  not try to re-use the MVIUsersController method. Copy what code you need to build a UserIdentity object and pass it to the MVI lookup method. If there's enough overlap, we can extract it out to a controller concern, that's the more proper way of sharing code vs. instantiating a controller and invoking a method on it.